syft/pkg/cataloger/cpe_test.go

```diff
@@ -1,1131 +0,0 @@
-		},
-	}
-	for _, test := range tests {
-		t.Run(strings.Join(test.input, ","), func(t *testing.T) {
-			val := strset.New(test.input...)
-			addSeparatorVariations(val)
-			assert.ElementsMatch(t, test.expected, val.List())
-		})
-	}
-}
-
-func Test_productsFromArtifactAndGroupIDs(t *testing.T) {
-	tests := []struct {
-		groupIDs   []string
-		artifactID string
-		expected   []string
-	}{
-		{
-			groupIDs:   []string{"org.sonatype.nexus"},
-			artifactID: "nexus-extender",
-			expected:   []string{"nexus", "nexus-extender"},
-		},
-		{
-			groupIDs: []string{"org.sonatype.nexus"},
-			expected: []string{"nexus"},
-		},
-		{
-			groupIDs:   []string{"org.jenkins-ci.plugins"},
-			artifactID: "ant",
-			expected:   []string{"ant"},
-		},
-		{
-			groupIDs:   []string{"org.jenkins-ci.plugins"},
-			artifactID: "antisamy-markup-formatter",
-			expected:   []string{"antisamy-markup-formatter"},
-		},
-		{
-			groupIDs:   []string{"io.jenkins.plugins"},
-			artifactID: "aws-global-configuration",
-			expected:   []string{"aws-global-configuration"},
-		},
-		{
-			groupIDs:   []string{"com.cloudbees.jenkins.plugins"},
-			artifactID: "cloudbees-servicenow-jenkins-plugin",
-			expected:   []string{"cloudbees-servicenow-jenkins-plugin"},
-		},
-		{
-			groupIDs:   []string{"com.atlassian.confluence.plugins"},
-			artifactID: "confluence-mobile-plugin",
-			expected:   []string{"confluence-mobile-plugin"},
-		},
-		{
-			groupIDs:   []string{"com.atlassian.confluence.plugins"},
-			artifactID: "confluence-view-file-macro",
-			expected:   []string{"confluence-view-file-macro"},
-		},
-		{
-			groupIDs:   []string{"com.google.guava"},
-			artifactID: "failureaccess",
-			expected:   []string{"failureaccess"},
-		},
-	}
-	for _, test := range tests {
-		t.Run(strings.Join(test.groupIDs, ",")+":"+test.artifactID, func(t *testing.T) {
-			actual := productsFromArtifactAndGroupIDs(test.artifactID, test.groupIDs)
-			assert.ElementsMatch(t, test.expected, actual, "different products")
-		})
-	}
-}
-
-func Test_candidateProductsForJava(t *testing.T) {
-	tests := []struct {
-		name     string
-		pkg      pkg.Package
-		expected []string
-	}{
-		{
-			name: "duplicate groupID in artifactID field",
-			pkg: pkg.Package{
-				Metadata: pkg.JavaMetadata{
-					PomProperties: &pkg.PomProperties{
-						GroupID:    "org.sonatype.nexus",
-						ArtifactID: "org.sonatype.nexus",
-					},
-				},
-			},
-			expected: []string{"nexus"},
-		},
-		{
-			name: "detect groupID-like value in artifactID field",
-			pkg: pkg.Package{
-				Metadata: pkg.JavaMetadata{
-					PomProperties: &pkg.PomProperties{
-						ArtifactID: "org.sonatype.nexus",
-					},
-				},
-			},
-			expected: []string{"nexus"},
-		},
-	}
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			actual := candidateProductsForJava(test.pkg)
-			assert.ElementsMatch(t, test.expected, actual, "different products")
-		})
-	}
-}
-
-func Test_vendorsFromGroupIDs(t *testing.T) {
```

Comment:
this needs to be cleaned up in the new java_test.go --note the duplicate cases

Reply:
cleaned up and simplified in 8d9c8b11ac254062c89e2a6362cc2bfd3589bf0b (removed sub selection generation since this is done downstream)